cpp/dolfinx/mesh/generation.h

```diff
@@ -264,61 +282,48 @@ std::vector<T> create_geom(MPI_Comm comm,
                            std::array<std::int64_t, 3> n)
 {
   // Extract data
-  const std::array<double, 3> p0 = p[0];
-  const std::array<double, 3> p1 = p[1];
-  std::int64_t nx = n[0];
-  std::int64_t ny = n[1];
-  std::int64_t nz = n[2];
-
-  const std::int64_t n_points = (nx + 1) * (ny + 1) * (nz + 1);
-  std::array range_p = dolfinx::MPI::local_range(
-      dolfinx::MPI::rank(comm), n_points, dolfinx::MPI::size(comm));
-
-  // Extract minimum and maximum coordinates
-  const double x0 = std::min(p0[0], p1[0]);
-  const double x1 = std::max(p0[0], p1[0]);
-  const double y0 = std::min(p0[1], p1[1]);
-  const double y1 = std::max(p0[1], p1[1]);
-  const double z0 = std::min(p0[2], p1[2]);
-  const double z1 = std::max(p0[2], p1[2]);
-
-  const T a = x0;
-  const T b = x1;
-  const T ab = (b - a) / static_cast<T>(nx);
-  const T c = y0;
-  const T d = y1;
-  const T cd = (d - c) / static_cast<T>(ny);
-  const T e = z0;
-  const T f = z1;
-  const T ef = (f - e) / static_cast<T>(nz);
-
-  if (std::abs(x0 - x1) < 2.0 * std::numeric_limits<double>::epsilon()
-      or std::abs(y0 - y1) < 2.0 * std::numeric_limits<double>::epsilon()
-      or std::abs(z0 - z1) < 2.0 * std::numeric_limits<double>::epsilon())
+  auto& [p0, p1] = p;
+  const auto [nx, ny, nz] = n;
+
+  assert(std::ranges::all_of(n, [](auto e) { return e >= 1; }));
+  for (std::int64_t i = 0; i < 3; i++)
```

Comment:
All these checks have been performed in the callers of `create_geom` so for double guarding we assert it once more, but do we really need to handle the exception once more?

Reply:
If it's already been check it could be skipped here.

A check at the highest level should raise an exception rather that test via an assert (I haven't checked what it's doing).

Reply:
Checked again, all code paths go through `create_box`, which raises an exception.